src/main/java/io/jenkins/plugins/audit/filter/RequestContextFilter.java

```diff
@@ -0,0 +1,35 @@
+package io.jenkins.plugins.audit.filter;
+
+import hudson.Extension;
+import hudson.init.Initializer;
+import hudson.init.Terminator;
+import hudson.util.PluginServletFilter;
+import jenkins.model.Jenkins;
+import org.apache.logging.log4j.ThreadContext;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import java.io.IOException;
+
+@Extension
+public class RequestContextFilter extends PluginServletFilter {
+
+    @Initializer
+    public static void init() throws ServletException {
+        PluginServletFilter.addFilter(new RequestContextFilter());
+    }
+
+    @Terminator
+    public void destroy() {
+        // Is this necessary
+    }
+
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        // I use the threadcontext here and as I have told you when building, it hangs and shuts down the machine.
```

Comment:
Here we need to populate the attributes that are common, right?

Reply:
It might be a good idea to reference/discuss this in the Gitter outreachy-chat so others can contribute.

Reply:
This is an area you would set the common attributes. For example:

```java
RequestContext.setSomething(something);
// ... and the others
chain.doFilter(request, response);
RequestContext.clear();
```

As for the conversation, GitHub is open for others as well, and it helps keep conversations close to the code in question here.